### PR TITLE
feat: #1463 Create modal to display App contact details

### PR DESCRIPTION
**Pull request by trankhacvy**
# Pull request checklist

## Does this close any currently open issues?

fixes: https://github.com/reapit/foundations/issues/1463

**Please check if your PR fulfills the following requirements:**

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/marketplace/src/components/pages/client-app-detail/contact-developer-section.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      >
+        NEED HELP?
+      </Button>
+
+      <Modal
+        visible={visible}
+        title={'Contact Details'}
+        afterClose={closeContactModal}
+        footerItems={
+          <>
```
Comment (tanphamhaiduong) [inline packages/marketplace/src/components/pages/client-app-detail/contact-developer-section.tsx:48]: There is only Button inside. Do we need <> ?

File: packages/marketplace/src/components/pages/client-app-detail/contact-developer-section.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  hasGutter?: boolean
+}
+
+export const ContactDeveloperSection = ({
+  contact: { developer, telephone, supportEmail, homePage },
+  hasGutter = true,
+}: ContactDeveloperSectionType) => {
+  const [visible, setVisible] = React.useState(false)
+
+  const openContactModal = () => {
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/client-app-detail/contact-developer-section.tsx:23]: Please add a test for this function

File: packages/marketplace/src/components/pages/client-app-detail/contact-developer-section.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    setVisible(true)
+  }
+
+  const closeContactModal = () => {
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/client-app-detail/contact-developer-section.tsx:27]: Please add a test for this function

File: packages/marketplace/src/components/pages/client-app-detail/client-app-detail.tsx [hunk 4]
```diff
@@ -181,10 +181,19 @@ const ClientAppDetail: React.FC<ClientAppDetailProps> = () => {
...
         <AppContent desktopIntegrationTypes={userDesktopIntegrationTypes} appDetailData={appDetailData} />
-        <FormSection className={classNames('is-clearfix', clientAppDetailStyles.footerContainer)}>
-          <Button className="is-pulled-right" onClick={onBackToAppsButtonClick(history, loginType)}>
-            Back To Apps
-          </Button>
+        <FormSection className={clientAppDetailStyles.footerContainer}>
+          <div className={clientAppDetailStyles.hiddenInDesktopScreenSize}>
+            <ContactDeveloperSection
+              contact={{
+                developer: appDetailData.developer,
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/client-app-detail/client-app-detail.tsx:188]: Please check line #160:
`const { id = '', installedOn = '' } = appDetailData`

Can we also unpack the values here using destructuring assignment syntax instead?

**Pull request merged**